rpc/client.go

```diff
@@ -462,6 +483,9 @@ func (c *Client) BatchCallContext(ctx context.Context, b []BatchElem) error {
 	for _, index := range byID {
 		elem := &b[index]
 		elem.Error = ErrMissingBatchResponse
+		if recordDone != nil {
+			recordDone[index](ctx, msgs[index], nil)
+		}
```

Comment:
If you wanted to DRY this a bit, you could make `c.MaybeUseRecorder(fn func)` to encapsulate the nil check. Not sure it's actually cleaner, but it would help us not miss nil checks maybe.